resources/js/modding-profile/main.tsx

```diff
@@ -0,0 +1,330 @@
+            <UserProfileContainer user={this.props.user}>
+              <HeaderV4
+                backgroundImage={this.props.user.cover.url}
+                links={headerLinks(this.props.user, 'modding')}
+                // add space for warning banner when user is blocked
+                modifiers={{
+                  restricted: core.currentUserModel.blocks.has(this.props.user.id) || this.props.user.is_restricted,
+                }}
+                theme='users'
+              />
+              <div className='osu-page osu-page--generic-compact'>
+                <div ref={this.pageRefs.main} data-page-id='main'>
+                  <Cover
+                    coverUrl={this.props.user.cover.url}
+                    currentMode={this.props.user.playmode}
+                    user={this.props.user}
+                  />
+                  {!this.props.user.is_bot && (
+                    <>
+                      <ProfileTournamentBanner banner={this.props.user.active_tournament_banner} />
+                      <div className='profile-detail'>
+                        <Badges badges={this.props.user.badges} />
+                        <Stats user={this.props.user} />
+                      </div>
+                    </>
+                  )}
+                  <DetailBar user={this.props.user} />
+                </div>
+                <div
+                  ref={this.pagesOffsetRef}
+                  className='hidden-xs page-extra-tabs page-extra-tabs--profile-page'
+                >
+                  <div
+                    ref={this.tabs}
+                    className='page-mode page-mode--profile-page-extra'
+                  >
+                    {moddingExtraPages.map((page) => (
+                      <a
+                        key={page}
+                        className='page-mode__item'
+                        data-page-id={page}
+                        href={`#${page}`}
+                        onClick={this.tabClick}
+                      >
+                        <ProfilePageExtraTab
+                          currentPage={this.currentPage}
+                          page={page}
+                        />
+                      </a>
+                    ))}
+                  </div>
+                </div>
+                <div ref={this.pages} className='user-profile-pages'>
+                  {moddingExtraPages.map((name) => (
+                    <div
+                      key={name}
+                      ref={this.pageRefs[name]}
+                      data-page-id={name}
+                    >
+                      {this.extraPage(name)}
+                    </div>
+                  ))}
+                </div>
+              </div>
+            </UserProfileContainer>
+          </BeatmapsContext.Provider>
+        </BeatmapsetsContext.Provider>
+      </DiscussionsContext.Provider>
+    );
+  }
+
+  private extraPage = (name: ModdingExtraPage) => {
```

Comment:
I think some of the components should be observer? ...or not? 🤔

Reply:
I wonder about that 🤔  (apparently everything that receives an `observable` prop should also be an `observer`, otherwise it's supposed to be `toJS`d even if the props don't change? 🤔 )

Reply:
yeah because otherwise when passing object and any of its properties get updated the non-observer component will be outdated. The toJS tells that the current component do access the properties so it'll update its children accordingly. Although in this case there's no update involved so it should be fine...? I'd probably still make them observer anyway just in case.